### DISCUSSION TITLE
Board completed

### DISCUSSION THREAD

**Discussion by tuffelh** [biologist79/ESPuino#60] [General]
Hallo,

ich habe das Board nun bestückt (V1). Es funktioniert auch alles, nur der RFID Leser funktioniert sehr schlecht.
Ich muss die Karten direkt auf den Leser halten, damit er die Karte liest. Irgendeine Idee an welcher Stellschraube ich da drehen kann?

Ansonsten noch ein Vorschlage:
Play, Pause, Titelauswahl auch über die Webui/http-request ermöglichen.

Ein Problem habe ich noch, mein WLAN hat ein recht komplexes Passwort, wenn ich dieses speichere und dann den Tonuino reboote, dann geht direkt der Hotspot wieder auf :(
In serial out kann ich aber sehen, dass das Passwort korrekt angezeigt wird.

**Comment by biologist79** [comment]
Moin,

die RC522 scheinen von der Qualität irgendwie unterschiedlich zu sein. Ich hatte auch schon welche, da musste man ziemlich nah ran und bei anderen ging es besser. Nutzt du Platformio? Ich frage nur deswegen, weil die RFID-Lib ja so eine Standard-Lib ist und zB in der Arduino-IDE vermutlich darüber eingebunden wird. Ich habe davon jedenfalls einen Fork gemacht und die Empfindlichkeit maximal erhöht. Siehe hier: https://github.com/biologist79/rfid/commit/cd626f46d6f556d0c7d086c42244d66901e859c3
Diese Lib habe ich in platformio.ini auch referenziert.

Also ich habe an der Stelle eigentlich nur zwei Tipps:
a) Es mal mit einem anderen Reader testen.
b) Es mal mit den PN5180 zu versuchen. Ist etwas teurer, aber die Reichweite ist erheblich besser. Nachteil ist, dass er mindestens zwei Pins mehr benötigt. "Mindestens" deswegen, weil ein Benutzer noch ein Feature entwickelt hat, bei dem man den Tonuino sogar aus dem Deepsleep aufwecken kann mit einer Karte. Dafür braucht es dann noch einen dritten Pin; muss man aber nicht nutzen das Feature. Das ist in meinem Master-Branch auch eh noch nicht enthalten, da ich das noch nicht final getestet habe.

Zum Passwort: Bist du weit von deinem WLAN-Router weg? Direkt nach dem Flashen habe ich zuweilen auch das Problem, dass der Tonuino nicht ins WLAN kommt (warum auch immer), aber danach geht es wieder. Hast mal versucht, die Zugangsdaten noch ein weiteres Mal einzutragen und zu rebooten?

VG.

**Reply by tuffelh** [reply]
Ich nutze Platform.io, sollte also deine lib sein. Dann versuche ich heute Abend mal nen weiteren 522er, drei hab ich noch ;)
Wenn alles nicht hilft, dann versuche ich es mit dem PN5180. Das aufwecken mit der RFID-Karte brauche ich aber vermutlich nicht.

Passwort: Habe etliche reboots versucht, nur im Gast-WLAN mit simplem Passwort hat es dann funktioniert. Sitze quasi 2m neben dem Router :-/
Das Passwort ist allerdings auch 63 Zeichen lang inkl. Sonderzeichen. (Würde ich so auch nicht nochmal machen :-D)

**Comment by biologist79** [comment]
Hmm, muss ich mal nachschauen, ob ich da irgendwelche Längenbeschränkungen eingebaut habe. Hast du mal andere Projekte verwendet, bei denen man die Zugangsdaten im Code fest einkodiert? Würde mich mal interessieren, ob es da geht.

**Reply by tuffelh** [reply]
Ich habe etliche esp8266 im Einsatz (shelly, wled, tasmota und esphome) bei denen funktioniert es.
Naja, wenn es nicht geht ist das nicht schlimm. Habe eh neue Accesspoints bestellt, dann bekommen die IOT-Sachen alle nen eigenes VLAN.
Hat übrigens mega Spaß gemacht das Ganze zu basteln. Mit deinem Board vor allem nochmals schöner als mit ner "fliegenden" Verdrahtung!
Wenn das RFID-Problem behoben ist und alles im Gehäuse ist, folgt dann die MQTT-Integration mit Home Assistant :)

**Comment by biologist79** [comment]
Hehe, das freut mich zu hören. Ja mit MQTT ist wirklich schick. Da habe ich in der Entwicklung viel Zeit reingesteckt, weil wir das auch viel nutzen. Der Sohnemann hat zB die Angewohnheit, seinen Tonuino (warum auch immer) etwas zu laut zu stellen. Kann ich dann ohne Diskussion leiser machen :-) Oder ich bringe ein Kind ins Bett und dimme den Neopixel runter, mache leiser, schalte aus oder aktiviere einen Sleepmodus.

Die Buttons zur Steuerung (aus deiner ersten Mail) wurden schon öfter mal gefordert. Wäre technisch auch gar nicht schwer. Zumindest dann, wenn man sich gut mit Javascript auskennt. Da ich mich immer einlesen muss und es auch via MQTT geht, hat sich meine Motivation bisher in Grenzen gehalten wenn ich ehrlich bin :-)

**Reply by tuffelh** [reply]
Also wenn ich erst MQTT eingerichtet habe wird es für mich wohl auch keine Rolle mehr spielen :-D
Aber erstmal das Gehäuse....und vielleicht noch ein anderer Lautsprecher. Der Visaton FR7 klingt doch etwas dünn.

**Comment by biologist79** [comment]
Ich würde bei einem kleinen Verstärker jetzt auch nicht zu viel erwarten. An 3.3V bringt der MAX ja nur 1,3W; an 5V sind es 3W. Denke da muss man schon mit anderen Verstärkern arbeiten, um da mehr rauszuholen.
Zu MQTT: Für MQTT selbst braucht es ja nur einen Broker; den kann man mit mosquitto einfach zB auf einem Raspi installieren. Fürs Handy gibt's auch MQTT-Apps. Vielleicht ist da auch was dabei, wo man sich eine kleine GUI konfigurieren kann.

**Reply by tuffelh** [reply]
Hatte garnicht daran gedacht, dass sich ja auch die Leistung des Verstärkers bei 3,3V reduziert. Naja ich teste da einfach nochmal.
Wegen MQTT, ich habe MQTT schon in Verbindung mit Home Assistant laufen, muss das dann nur in die GUI einbauen, aber das macht natürlich erst Sinn wenn ich fertig bin und die neuen Access Points da sind.
Wenn ich es in der GUI habe, kann ich ja mal die Screenshots posten :)

**Comment by biologist79** [comment]
Ja, gerne. Ich kenne bislang nur openHAB. Sollte vielleicht mal Screenshots in mein Repository legen. Hat ja sonst keiner ne Vorstellung davon, was man da machen kann :-)
